config/config_test.go

```diff
@@ -2437,9 +2440,9 @@ func TestScrapeConfigDisableCompression(t *testing.T) {
 
 func TestScrapeConfigNameValidationSettings(t *testing.T) {
 	model.NameValidationScheme = model.UTF8Validation
-	defer func() {
+	t.Cleanup(func() {
```

Comment:
Is there any need to change from `defer` into `t.Cleanup`? Otherwise, I think it's out of scope for this PR.

Reply:
Using `t.Cleanup` ensures the global state (`model.NameValidationScheme`) is reset immediately after each test, avoiding interference between parallel subtests (`t.Parallel`). `defer` would delay cleanup until the parent test completes, causing potential test flakiness. In our case, the test was failing.

```
# using defer
❯ go test ./config -count=1                  
--- FAIL: TestScrapeConfigNameValidationSettings (0.00s)
    --- FAIL: TestScrapeConfigNameValidationSettings/local_setting (0.00s)
panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 [recovered]
	panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 [recovered]
	panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8
```

Reply:
OK, thanks for explaining.